src/go/rpk/pkg/cli/cmd/status.go

```diff
@@ -149,44 +153,45 @@ func getMetrics(
 	timeout time.Duration,
 	conf config.Config,
 	send bool,
-	out chan<- [][]string,
-) {
-	rows := [][]string{}
+) (*metricsResult, error) {
+	res := &metricsResult{[][]string{}, nil}
 	osInfo, err := system.UnameAndDistro(timeout)
 	if err != nil {
-		log.Info("Error querying OS info: ", err)
-	} else {
-		rows = append(rows, []string{"OS", osInfo})
+		return res, fmt.Errorf("Error querying OS info: %v", err)
 	}
+	res.rows = append(res.rows, []string{"OS", osInfo})
+
 	cpuInfo, err := system.CpuInfo()
 	if err != nil {
-		log.Info("Error querying CPU info: ", err)
+		return res, fmt.Errorf("Error querying CPU info: %v", err)
 	}
 	cpuModel := ""
 	if len(cpuInfo) > 0 {
 		cpuModel = cpuInfo[0].ModelName
-		rows = append(rows, []string{"CPU Model", cpuModel})
+		res.rows = append(res.rows, []string{"CPU Model", cpuModel})
 	}
 	m, errs := system.GatherMetrics(fs, timeout, conf)
 	if len(errs) != 0 {
+		msgs := ""
 		for _, err := range errs {
-			log.Debugf("Error gathering metrics: %v", err)
+			msgs = fmt.Sprintf("%s %v.", msgs, err)
 		}
-	} else {
-		rows = append(
-			rows,
-			[]string{"CPU Usage %", fmt.Sprintf("%0.3f", m.CpuPercentage)},
-			[]string{"Free Memory (MB)", fmt.Sprintf("%0.3f", m.FreeMemoryMB)},
-			[]string{"Free Space  (MB)", fmt.Sprintf("%0.3f", m.FreeSpaceMB)},
-		)
-	}
+		return res, errors.Errorf("Error gathering metrics:%s", msgs)
```

Comment:
```diff
-		return res, errors.Errorf("Error gathering metrics:%s", msgs)
+		return res, errors.Errorf("Error gathering metrics: %s", msgs)
```

Reply:
Each message adds its leading space in https://github.com/vectorizedio/redpanda/pull/329/commits/e7fba2ddbe1a02a089b5f5742ae368ba2d3df443#diff-d5153c54fde06d72d3bc5c53e8c3bf834ed8ea990c3a9f47418ec94cf56aaefcR177

It's kind of confusing. I could also "map" (cries in Go) the list to get a list with only the error strings and then `strings.Join` them too.

Reply:
Maybe it's best to take a more structure approach, e.g. [errors.Wrap](https://godoc.org/github.com/pkg/errors#Wrap) and [hashicorp/go-multierror](https://github.com/hashicorp/go-multierror)

Reply:
I thought of `errors.Wrap`, which works, but it's meant to _add context_ to an error, not to represent a list of errors.
I had forgotten about multierror, Hashicorp always has something for your Go needs haha.

Reply:
They both help with formatting, though, which should vastly reduce the string manipulation required.